app/assets/javascripts/discourse/app/routes/new-message.js

```diff
@@ -12,49 +12,74 @@ export default DiscourseRoute.extend({
 
   beforeModel(transition) {
     const params = transition.to.queryParams;
-
+    const userName = params.username;
     const groupName = params.groupname || params.group_name;
 
-    if (this.currentUser) {
+    if (!this.currentUser) {
+      cookie("destination_url", window.location.href);
+      this.router.replaceWith("login");
+      return;
+    }
+
+    if (transition.from) {
+      transition.abort();
+
+      if (userName) {
+        return this.openComposer(transition, userName);
+      }
+
+      if (groupName) {
+        // send a message to a group
+        return Group.messageable(groupName)
+          .then((result) => {
+            if (result.messageable) {
+              this.openComposer(transition, groupName);
+            } else {
+              this.dialog.alert(
+                I18n.t("composer.cant_send_pm", { username: groupName })
+              );
+            }
+          })
+          .catch(() => this.dialog.alert(I18n.t("generic_error")));
```

Comment:
I know you just moved code around, but any chances we could provide the user with a better error message than a generic one? 🤔

Reply:
Should we just pass along the error message then? Not sure what specifics we can provide here because even we're not sure what's the error in this case.

Note just above we're giving clear error message `I18n.t("composer.cant_send_pm", { username: groupName })` because we know that the group is not messageable, but in this case this just might be network error, hence we're falling back to generic error.

Maybe a better worded message like: "Sorry, there was an error creating that message. Please try again."?

Reply:
If we truly don't know what happened, then sure, improving the generic message could help 👍